service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DiaryPlanStatementMapper.java

```diff
@@ -0,0 +1,147 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import java.util.Date;
+import java.util.Optional;
+import java.util.function.BinaryOperator;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DiaryPlanStatementMapper {
+
+    private static final Mustache PLAN_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_plan_statement_template.mustache");
+    private static final String COMMA = ", ";
+    private static final String EMPTY_DATE = "nullFlavor=\"UNK\"";
+    private static final String FULL_DATE = "value=\"%s\"";
+
+    private final MessageContext messageContext;
+
+    public String mapDiaryProcedureRequestToPlanStatement(ProcedureRequest procedureRequest, Boolean isNested) {
+        if (procedureRequest.getIntent() != ProcedureRequest.ProcedureRequestIntent.PLAN) {
+            return StringUtils.EMPTY;
+        }
+
+        PlanStatementMapperParameters.PlanStatementMapperParametersBuilder builder = PlanStatementMapperParameters.builder()
+            .isNested(isNested)
+            .id(messageContext.getIdMapper().getOrNew(ResourceType.ProcedureRequest, procedureRequest.getId()))
+            .availabilityTime(buildAvailabilityTime(procedureRequest));
+
+        buildEffectiveTime(procedureRequest).map(builder::effectiveTime);
+        buildText(procedureRequest).map(builder::text);
+
+        return TemplateUtils.fillTemplate(PLAN_STATEMENT_TEMPLATE, builder.build());
+    }
+
+    private Optional<String> buildEffectiveTime(ProcedureRequest procedureRequest) {
+        Date date = null;
+        if (procedureRequest.hasOccurrenceDateTimeType()) {
+            date = procedureRequest.getOccurrenceDateTimeType().getValue();
+        } else if (procedureRequest.hasOccurrencePeriod()) {
```

Comment:
I think the part of the occurrencePeriod mapping that also prepends to text is missing. See speadsheet for full details.

>  if occurrencePeriod.start then form  a string from the occurrencePeriod.start like 'Earliest Recall Date: {occurrencePeriod.start} " and prepend it to the contents of the <text> element

Reply:
Just added in line 90, when building a note.